Mathlib/Tactic/Alias.lean

```diff
@@ -0,0 +1,171 @@
+/-
+Copyright (c) 2017 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+import Lean.Elab.Command
+import Lean.Elab.Term
+import Lean
+
+/-!
+# The `alias` command
+
+This file defines an `alias` command, which can be used to create copies
+of a theorem or definition with different names.
+
+Syntax:
+
+```lean
+/-- doc string -/
+alias my_theorem ← alias1 alias2 ...
+```
+
+This produces defs or theorems of the form:
+
+```lean
+/-- doc string -/
+theorem alias1 : <type of my_theorem> := my_theorem
+
+/-- doc string -/
+theorem alias2 : <type of my_theorem> := my_theorem
+```
+
+Iff alias syntax:
+
+```lean
+alias A_iff_B ↔ B_of_A A_of_B
+alias A_iff_B ↔ ..
+```
+
+This gets an existing biconditional theorem `A_iff_B` and produces
+the one-way implications `B_of_A` and `A_of_B` (with no change in
+implicit arguments). A blank `_` can be used to avoid generating one direction.
+The `..` notation attempts to generate the 'of'-names automatically when the
+input theorem has the form `A_iff_B` or `A_iff_B_left` etc.
+-/
+
+namespace Tactic
+namespace Alias
+
+open Lean
+
+/-- Adds some copies of a theorem or definition. -/
+syntax (name := alias) "alias " ident " ← " ident* : command
+
+/-- Adds one-way implication declarations. -/
+syntax (name := aliasLR) "alias " ident " ↔ " binderIdent binderIdent : command
+
+/-- Adds one-way implication declarations, inferring names for them. -/
+syntax (name := aliasLRDots) "alias " ident " ↔ " ".." : command
+
+/-- Elaborates an `alias ←` command. -/
+@[commandElab «alias»] def elabAlias : Elab.Command.CommandElab
+| `(alias $name:ident ← $aliases:ident*) => do
+  let resolved ← resolveGlobalConstNoOverload name
+  let constant ← match (← Lean.getEnv).constants.find? resolved with
+                 | none => throwError "failed to resolve alias LHS"
+                 | some c => pure c
+
+  for a in aliases do
+    let decl ← match constant with
+    | Lean.ConstantInfo.defnInfo d =>
+      pure $ Lean.Declaration.defnDecl {
+        d with name := a.getId
+               value := mkConst resolved
```

Comment:
```suggestion
               value := mkConst resolved (d.levelParams.map mkLevelParam)
```
Otherwise you can't alias `def foo : Type u := Sort u` (same applies below).